### ISSUE TITLE
Unable to download above 16 44.1

### ISSUE THREAD

**Issue by Avkash76** [cstaelen/tidarr#26]
Hello, I can't download lossless albums.

In the configuration folder, if I leave "hifi" it downloads max in 16 44.1.

If I change to "HI_RES_LOSSLESS" or "LOSSLESS", it downloads M4A at 96kb/s, worse than MP3.

Is the project abandoned and is that normal?

Thank you so much

**Comment by cstaelen** [comment]
Hello, 

The project is still active, even slowly, as I'm the only maintainer. 
With a simple individual plan subscription to Tidal, I can't get more than FLAC file 16 44.1 too.

As Tidarr use TidalMediaDownloader as dependency, IMO this is much more a question to ask here : 
https://github.com/yaronzz/Tidal-Media-Downloader

Also, I know there is TIDDL ,seems to be more active : https://github.com/oskvr37/tiddl 
But i didn't try yet. Do you know if you can get higher quality ?

**Comment by cstaelen** [comment]
Have you tried `Master`  as quality value ?
https://doc.yaronzz.com/post/tidal_dl_installation/

Available quality values seems to be : `Normal`/`High`/`HiFi`/`Master`

You can set the quality param in `config/.tidal-dl.json`  file, and restart container. 

Regards,

**Comment by cstaelen** [comment]
It seems possible to download max quality 24 bit 96.0 kHz with Tiddl script 🙌  
Stay tuned for the next release !

**Comment by cstaelen** [comment]
Hello, 

For the record I just release 0.1.3 tag. 
You can now download at 24bit 96khz (if available) by adding a new env var in you `docker-compose.yml`:
```
 environment:
  - ...
  - TIDDL_QUALITY=master
```

Let me know if all is fine on your side 👍 

Regards,

**Comment by Avkash76** [comment]
Hi cstaelen,

I just updated the image and every time I want to download an album, it crashes.

=== Tiddl ===
Executing: 166790187 -q master -p /home/app/standalone/download/incomplete -o "{artist}/{album}/{title}" -s
Tiddl: Traceback (most recent call last):
  File "/usr/bin/tiddl", line 8, in <module>

Tiddl:     sys.exit(main())
             ^^^^^^
  File "/usr/lib/python3.11/site-packages/tiddl/__init__.py", line 256, in main
    downloadAlbum(input_id, skip_existing)
  File "/usr/lib/python3.11/site-packages/tiddl/__init__.py", line 204, in downloadAlbum
    logger.info(f"album: {album['title']}")
                          ~~~~~^^^^^^^^^

Tiddl: KeyError: 'title'

Tiddl process exited with code 1


I use this docker, so my father can easily download his music with our Tidal account. Personally I use TIDAL-Downloader-NG which allows you to go up to 192khz.

Thanks for the update and your quick response.

**Comment by cstaelen** [comment]
Hi, 

Thanks for your feedback !

It seems to be an issue with country code.
This track may not be available for your country (it depends on your Tidal account).

Could you please check your account country code :
```bash
docker exec tidarr cat /root/.tiddl_config.json
```

Once you get it, you should set country code as env var in your `docker-compose.yml`

```yaml
 environment:
  - ...
  - REACT_APP_TIDAL_COUNTRY_CODE=<country-code>
```

Thank you for your help :)

**Comment by cstaelen** [comment]
Indeed, I just checked with https://tidal.com/browse/album/14234194
I can download it at `192khz` 👍 

Terminal output :
`[ TIDDL ] Hurricane :: Max Quality - 24 bit 192.0 kHz`

**Comment by Avkash76** [comment]
I changed the country and everything works perfectly. Thank you very much for your help. My father will be happy.

One last question, is it possible to add a password to the interface? Via the config file perhaps?

Thank you so much

**Comment by cstaelen** [comment]
At the moment, there is no easy way to add an auth form (maybe with traefik or caddy ?).
But it's a good idea for the next feature indeed 👍 

And thank to you to pointing those enhancements out !

**Comment by cstaelen** [comment]
By the way, when using quality `master`, 96 and 192 tracks often have `m4e`file extension.
In 0.1.4, you can set the desired extension as env var to encode it : 
```yaml
environment:
  - TIDDL_FORCE_EXT=<flac|mp3|m4a> # default: unset, depending the track downloaded.
```

**Comment by Avkash76** [comment]
Great ! For the conversion to flac I imagine you have to add FFmpeg in the config folder or define an address in the config file?

**Comment by cstaelen** [comment]
Yes indeed, I had to `apk add ffmpeg` in the docker container 👍 
But there is no additional configuration for that expect the env var `TIDDL_FORCE_EXT`.

**Comment by cstaelen** [comment]
Hello @Avkash76,  

I just released 0.1.7 version with new features you may be interested in, including auth form :) 

You should be able to add an authentication form with this env var `- ADMIN_PASSWORD=123pwd`

Hope you will enjoy :)

**Comment by Avkash76** [comment]
Hello cstaelen,

Wow perfect, this is exactly what I wanted. Thank you so much.

Avkash
